cmake/CompilerWarnings.cmake

```diff
@@ -67,6 +67,7 @@ function(set_project_warnings project_name)
           -Wuseless-cast # warn if you perform a cast to the same type
           -Wno-interference-size # suppress ABI compatibility warnings for hardware inferred size
           -Wno-maybe-uninitialized # false positives if asan is enabled: https://gcc.gnu.org/bugzilla//show_bug.cgi?id=1056h6
+          -fconcepts-diagnostics-depth=3
```

Comment:
@RalphSteinhagen I'm fine with keeping this as a local change. Let me know if you want it.